processing/src/main/java/org/apache/druid/frame/processor/SuperSorter.java

```diff
@@ -761,12 +865,42 @@ private long getTotalMergersInLevel(final int level)
     } else if (level >= totalMergingLevels) {
       throw new ISE("Invalid level %d", level);
     } else if (level == totalMergingLevels - 1) {
-      return outputPartitionsFuture.isDone() ? getOutputPartitions().size() : UNKNOWN_TOTAL;
+      if (outputPartitionsFuture.isDone()) {
+        return totalInputFrames == 0 ? 0 : getOutputPartitions().size();
+      } else {
+        return UNKNOWN_TOTAL;
+      }
+    } else if (level > 0 && level == totalMergingLevels - 2) {
```

Comment:
`isPenultimateLevel` might make sense; I think it'd have to return a tri-state since the answer might be unknown. I/we can think about it in a future patch— simplifying the logic of SuperSorter would be great.